controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -116,7 +119,7 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	// Check if the HostedControlPlane is ready
 	if !hostedcontrolplane.Status.Ready {
 		log.Info("skipped deploying monitoring objects: HostedControlPlane not ready")
-		utilreconcile.RequeueWith(nil)
+		return utilreconcile.RequeueWith(nil)
```

Comment:
Passing `nil` here is misleading - https://github.com/openshift/route-monitor-operator/blob/master/pkg/util/reconcile/reconcile_wrapper.go#L25-L31 it doesn't actually trigger a reconcile. I'm not sure about this whole controller, but specifically here I think we want to pass something like this instead
```suggestion
		return utilreconcile.RequeueWith(errors.New("skipped deploying monitoring objects: HostedControlPlane not ready"))
```